pyingest/parsers/jats.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python
+
+import bs4
+from default import BaseBeautifulSoupParser
+from pyingest.config.config import *
+
+
+class NoSchemaException(Exception):
+    pass
+
+class WrongSchemaException(Exception):
+    pass
+
+class UnparseableException(Exception):
+    pass
+
+class JATSParser(BaseBeautifulSoupParser):
+
+    def __init__(self):
+        pass
+
+    def _detag(self, r, tags_keep, **kwargs):
+        newr = bs4.BeautifulSoup(unicode(r),'lxml')
+        tag_list = list(set([x.name for x in newr.find_all()]))
+        for t in tag_list:
+            if t in JATS_TAGS_DANGER:
+                newr.find(t).decompose()
+            elif t in tags_keep:
+                newr.find(t).contents
+            else:
+                newr.find(t).unwrap()
+        return unicode(newr).replace("  "," ")
+
+    def resource_dict(self, fp, **kwargs):
```

Comment:
Unnecessarily complex?  This should accomplish the same result:
```
   parsed_tree = self.bsfiletodict(fp, **kwargs)
   return parsed_tree.article
```

Reply:
I did that because I want to return a BeautifulSoup object rather than a Tag or NavigableString which is what returning object.article would do.  There are a few methods that only work with BeautifulSoup objects.

Reply:
Got it